examples/ios/objc/Draw/AppDelegate.m

```diff
@@ -0,0 +1,84 @@
+
+    application.applicationSupportsShakeToEdit = YES;
+    
+    // Setup Global Error Handler
+    RLMSyncErrorReportingBlock globalErrorHandler = ^(NSError *error, RLMSyncSession *session) {
+        NSLog(@"A global error has occurred! %@", error);
+    };
+    
+    [RLMSyncManager sharedManager].errorHandler = globalErrorHandler;
+    
+    if ([RLMSyncUser allUsers].count > 0) {
+        NSURL *syncURL = [NSURL URLWithString:[NSString stringWithFormat:@"realm://%@:9080/~/Draw", ipAddress]];
+        RLMSyncConfiguration *syncConfig = [[RLMSyncConfiguration alloc] initWithUser:[RLMSyncUser currentUser] realmURL:syncURL];
+        RLMRealmConfiguration *defaultConfig = [RLMRealmConfiguration defaultConfiguration];
+        defaultConfig.syncConfiguration = syncConfig;
+        [RLMRealmConfiguration setDefaultConfiguration:defaultConfig];
+    }
+    else {
+        // The base server path
+        // Set to connect to local or online host
+        NSURL *authURL = [NSURL URLWithString:[NSString stringWithFormat:@"http://%@:9080", ipAddress]];
+        
+        // Creating a debug credential since this demo is just using the generated access token
+        // produced when running the Realm Object Server via the `start-object-server.command`
+        RLMSyncCredentials *credential = [RLMSyncCredentials credentialsWithUsername:@"demo@realm.io"
+                                                                         password:@"password"
+                                                                          register:NO];
+        
+        // Log the user in (async, the Realm will start syncing once the user is logged in automatically)
+        [RLMSyncUser logInWithCredentials:credential
+                            authServerURL:authURL
+                             onCompletion:^(RLMSyncUser *user, NSError *error) {
+           if (error) {
+               NSLog(@"A login error has occurred! %@", error);
+           }
+           else { // Logged in setup the default Realm
+               // The Realm virtual path on the server.
+               // The `~` represents the Realm user ID. Since the user ID is not known until you
+               // log in, the ~ is used as short-hand to represent this.
+               NSURL *syncURL = [NSURL URLWithString:[NSString stringWithFormat:@"realm://%@:9080/~/Draw", ipAddress]];
+               RLMSyncConfiguration *syncConfig = [[RLMSyncConfiguration alloc] initWithUser:user realmURL:syncURL];
+               RLMRealmConfiguration *defaultConfig = [RLMRealmConfiguration defaultConfiguration];
+               defaultConfig.syncConfiguration = syncConfig;
+               [RLMRealmConfiguration setDefaultConfiguration:defaultConfig];
+           }
+        }];
+    }
+
+    self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
+    self.window.rootViewController = [[UIViewController alloc] init];
+    self.window.rootViewController.view = [DrawView new];
```

Comment:
The biggest problem with the way the code is currently structured is that if the Realm is accessed before the sync login callback is invoked, the default (local) Realm will be accessed first, and then it will be swapped out for the synced Realm when the callback is invoked at a later time.

We either need to not set `self.window.rootViewController.view` until the sync realm configuration is set, or we refactor `DrawView` to not access the Realm until we've set the default configuration.